galley/pkg/config/analysis/analyzers/util/find_errorline_utils.go

```diff
@@ -0,0 +1,169 @@
+	keyPath := "{.spec.ports[" + fmt.Sprintf("%d", portIndex) + "].port" + "}"
+	line := FindErrorLine(keyPath, r.Origin.GetFieldMap())
+	return line
+}
+
+// ErrorLineForFromRegistry returns the line number of fromRegistry in the mesh networks
+func ErrorLineForFromRegistry(networkName string, endPointIndex int, r *resource.Instance) int {
+	keyPath := "{.networks." + networkName + ".endpoints[" + fmt.Sprintf("%d", endPointIndex) + "]}"
+	line := FindErrorLine(keyPath, r.Origin.GetFieldMap())
+	return line
+}
+
+// ErrorLineForContainerImage returns the line number of the image in the container
+func ErrorLineForContainerImage(containerIndex int, r *resource.Instance) int {
+	keyPath := "{.spec.containers[}" + fmt.Sprintf("%d", containerIndex) + "].image}"
+	line := FindErrorLine(keyPath, r.Origin.GetFieldMap())
+	return line
+}
+
+// ErrorLineForMetaDataNameSpace returns the line number of the metadata.namespace
+func ErrorLineForMetaDataNameSpace(r *resource.Instance) int {
+	keyPath := "{.metadata.namespace}"
+	line := FindErrorLine(keyPath, r.Origin.GetFieldMap())
+	return line
+}
+
+// ErrorLineForMetaDataName returns the line number of the metadata.name
+func ErrorLineForMetaDataName(r *resource.Instance) int {
+	keyPath := "{.metadata.name}"
+	line := FindErrorLine(keyPath, r.Origin.GetFieldMap())
+	return line
+}
+
+// ErrorLineForAuthorizationPolicyNameSpace returns the line number of the namespaces for authorizationPolicy
+func ErrorLineForAuthorizationPolicyNameSpace(ruleIndex int, fromIndex int, namespaceIndex int, r *resource.Instance) int {
+	pathKey := "{.spec.rules[" + fmt.Sprintf("%d", ruleIndex) + "].from[" + fmt.Sprintf("%d", fromIndex) +
+		"].source.namespaces[" + fmt.Sprintf("%d", namespaceIndex) + "]}"
+	line := FindErrorLine(pathKey, r.Origin.GetFieldMap())
+	return line
+}
+
+// FinedErrorLineNum returns the error line number with the annotation name and the field map from resource.Origin as the input
+func ErrorLineForAnnotation(annotationName string, r *resource.Instance) int {
+	path := "{.metadata.annotations"
+	path += "." + annotationName + "}"
+	line := FindErrorLine(path, r.Origin.GetFieldMap())
+	return line
+}
+
+// ErrorLineForGatewaySelector returns the path of the gateway selector
+func ErrorLineForGatewaySelector(gwSelector labels.Selector, r *resource.Instance) int {
+	pathKey := "{.spec.selector"
+	selectorLabel := FindLabelForSelector(gwSelector)
+	pathKey += "." + selectorLabel + "}"
+	line := FindErrorLine(pathKey, r.Origin.GetFieldMap())
+	return line
+}
+
+// ErrorLineForCredentialName returns the path of the gateway selector
+func ErrorLineForCredentialName(serverIndex int, r *resource.Instance) int {
+	pathKey := "{.spec.servers[" + fmt.Sprintf("%d", serverIndex) + "].tls.credentialName}"
+	line := FindErrorLine(pathKey, r.Origin.GetFieldMap())
+	return line
+}
+
+// FindErrorLine returns the line number of the input key from the input map
+func FindErrorLine(key string, m map[string]int) int {
```

Comment:
As noted above, this should return (int, bool). In that case just
`return FindErrorLine(...)`
in the other functions will return the right values.